torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -805,6 +813,77 @@ struct NCCLTraceBuffer {
     }
   }
 
+  std::list<json> getCollectiveTraceJson(bool onlyActive) {
+    auto result = dump_entries();
+
+    std::list<json> entries;
+    for (auto i : c10::irange(result.size())) {
+      json j;
+      auto& e = result.at(i);
+      if (onlyActive && e.time_discovered_completed_.has_value()) {
+        continue;
+      }
+      j[record_id_key_str] = int64_t(e.id_);
```

Comment:
just curious- is it not easy to get an ivalue as a str() via some method?  I guess its still nicer the way you did it.

Reply:
Yes, not easy since `ivalue` is a `generic`. I'd have to write a lot of code in `ivalue` to make it happen.